services/identity/UsersService.ts

```diff
@@ -29,15 +26,28 @@ class UsersService {
 
   private readonly repository: UsersServiceProps["repository"]
 
-  private readonly whitelistDomains: string[]
+  private readonly sequelize: UsersServiceProps["sequelize"]
+
+  readonly whitelistDomains: string[]
 
-  constructor({ otp, mailer, smsClient, repository }: UsersServiceProps) {
+  constructor({
+    otp,
+    mailer,
+    smsClient,
+    repository,
+    sequelize,
+  }: UsersServiceProps) {
     this.otp = otp
     this.mailer = mailer
     this.smsClient = smsClient
     this.repository = repository
+    this.sequelize = sequelize
+
+    // Allowed domains is a semicolon separate list of domains (e.g. .gov.sg; @agency.com.sg; etc)
+    // that are allowed to login.
+    const { DOMAIN_WHITELIST } = process.env
 
-    this.whitelistDomains = (DOMAIN_WHITELIST || ".gov.sg")
+    this.whitelistDomains = (DOMAIN_WHITELIST || "gov.sg")
```

Comment:
The default value should be `.gov.sg`. Using `gov.sg` could open us to an attack where a malicious actor buys `evilgov.sg` and is able to use our system.

Additionally, we don't foresee users having `@gov.sg` domains since that is reserved for MCI, and MCI uses `@mci.gov.sg`.